### ISSUE TITLE
🐛 Warning when trying to use `biome-ignore-all assist/source/organizeImports`

### ISSUE THREAD

**Issue by iFonny** [biomejs/biome#7130]
### Environment information

```block
CLI:
  Version:                      2.1.3
  Color support:                true

Platform:
  CPU Architecture:             aarch64
  OS:                           macos

Environment:
  BIOME_LOG_PATH:               unset
  BIOME_LOG_PREFIX_NAME:        unset
  BIOME_CONFIG_PATH:            unset
  BIOME_THREADS:                unset
  NO_COLOR:                     unset
  TERM:                         xterm-256color
  JS_RUNTIME_VERSION:           v22.16.0
  JS_RUNTIME_NAME:              node
  NODE_PACKAGE_MANAGER:         npm/11.4.2

Biome Configuration:
  Error:                        The configuration schema version does not match the CLI version 2.1.3
  Status:                       Loaded with errors
  Path:                         biome.json
  Formatter enabled:            true
  Linter enabled:               true
  Assist enabled:               true
  VCS enabled:                  true

Workspace:
  Open Documents:               0
```

### What happened?

When using `biome lint` I get this warning (Suppression comment has no effect because the tool is not enabled)
I don't understand why I get this warning. If I remove the biome-ignore-all, biome will reorder my imports, which is not what I want for this file.   

```
> biome lint . --write

lib/main.ts:1:5 suppressions/unused ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━

  ⚠ Suppression comment has no effect because the tool is not enabled.
  
  > 1 │ /** biome-ignore-all assist/source/organizeImports: For the lib root file, we don't want to organize exports */
      │     ^^^^^^^^^^^^^^^^
```

My main.ts

```ts
/** biome-ignore-all assist/source/organizeImports: For the lib root file, we don't want to organize exports */

export function hello(thing: string): string {
  return `Hello ${thing}!`;
}

// --- Providers ---
export { DesignSystemProvider } from './providers/DesignSystemProvider.tsx';

// --- Atoms ---
export { Button } from './components/atoms/Button';
export type { ButtonProps } from './components/atoms/Button';

// --- Molecules ---

// --- Organisms ---
export { Table } from './components/organisms/Table';

// --- Templates ---

// --- Pages ---

// --- Types ---
export type * from './types';
```

biome.json:
```json
{
  "$schema": "https://biomejs.dev/schemas/2.1.3/schema.json",
  "vcs": {
    "enabled": true,
    "clientKind": "git",
    "useIgnoreFile": true
  },
  "files": {
    "ignoreUnknown": false
  },
  "formatter": {
    "enabled": true,
    "indentStyle": "space",
    "lineWidth": 120
  },

  "linter": {
    "enabled": true,
    "rules": {
      "recommended": true,
      "correctness": {
        "useExhaustiveDependencies": "warn"
      },
      "suspicious": {
        "noExplicitAny": "error",
        "noArrayIndexKey": "error",
        "noConsole": {
          "level": "warn",
          "options": {
            "allow": ["assert", "error", "info", "warn"]
          }
        }
      }
    }
  },

  "assist": {
    "enabled": true,
    "actions": {
      "source": {
        "organizeImports": "on"
      }
    }
  },

  "javascript": {
    "formatter": {
      "quoteStyle": "single"
    }
  },

  "overrides": [
    {
      "includes": ["**/*.test.{ts,tsx}", "**/*.spec.{ts,tsx}"],
      "linter": {
        "rules": {
          "suspicious": {
            "noConsole": "off"
          }
        }
      }
    },
    {
      "includes": ["**/*.d.ts"],
      "linter": {
        "rules": {
          "correctness": {
            "noUnusedVariables": "off"
          }
        }
      }
    }
  ]
}
```

### Expected result

I should not show a warning

### Code of Conduct

- [x] I agree to follow Biome's Code of Conduct

**Comment by ematipico** [comment]
The problem is that `biome lint` doesn't run the assist, and when the analyser sees a suppression comment that doesn't belong to the linter, it emits a warning.

Maybe we should remove that diagnostic altogether. I was trying to be smart :)

**Comment by sankargorthi** [comment]
happening to me with `// biome-ignore assist/source/useSortedKeys: page params must come after query fn for type inference`

`check` passes, but `lint` complains when run independently. Will wait for a fix. thanks!
